.github/workflows/docs.yaml

```
@@ -0,0 +1,49 @@
+name: Docs
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  docs:
+    runs-on: ubuntu-22.04
+    name: Build and validate documentation
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          toolchain: 1.81.0
+
+      - name: "Cache cargo"
+        id: cache-cargo
+        uses: "actions/cache@v4"
+        with:
+          path: |
+            ~/.cargo
+          save-always: true
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Install mdBook and plugins
+        run: |
+          cargo install --locked mdbook@0.4.40 mdbook-mermaid@0.15.0 mdbook-linkcheck@0.7.7
```

add new line for readability:
```suggestion
          cargo install --locked \
          mdbook@0.4.40 \
          mdbook-mermaid@0.15.0 \
          mdbook-linkcheck@0.7.7
```